src/i2s/std.rs

```diff
@@ -800,279 +743,85 @@ impl<'d, Dir: I2sRxSupported + I2sTxSupported> I2sStdDriver<'d, Dir> {
 
         // Safety: &pin_cfg is a valid pointer to an i2s_pin_config_t.
         unsafe {
-            esp!(i2s_set_pin(port, &pin_cfg))?;
+            esp!(i2s_set_pin(I2S::port(), &pin_cfg))?;
         }
 
-        Ok(Self {
-            i2s: port as u8,
-            _p: PhantomData,
-            _dir: PhantomData,
-        })
+        Ok(this)
     }
 }
 
-impl<'d, Dir: I2sRxSupported> I2sStdDriver<'d, Dir> {
-    /// Create a new standard mode driver for the given I2S peripheral with only the receive channel open.
-    #[cfg(not(esp_idf_version_major = "4"))]
+impl<'d> I2sDriver<'d, I2sBiDir> {
+    /// Create a new standard mode driver for the given I2S peripheral with both the receive and transmit channels open.
     #[allow(clippy::too_many_arguments)]
-    pub fn new_rx<I2S: I2s>(
-        _i2s: impl Peripheral<P = I2S> + 'd,
-        config: config::StdConfig,
+    pub fn new_std_bidir<I2S: I2s>(
+        i2s: impl Peripheral<P = I2S> + 'd,
+        config: &config::StdConfig,
         bclk: impl Peripheral<P = impl InputPin + OutputPin> + 'd,
-        din: Option<impl Peripheral<P = impl InputPin> + 'd>,
+        din: impl Peripheral<P = impl InputPin> + 'd,
```

Comment:
Similar to my comment on the TDM, there was a reason I had this an an `Option<>`, likely related to unfinished ADC work that is unlikely to come to fruition.

Reply:
How about we leave it non-optional. If we get to implementing the ADC mode, we can make this optional again?

Reply:
Sounds good to me!